Fix EDDN Market Create

nixpulvis opened the pull request:
EDDN commodity messages give `systemName`, `stationName` and `marketId`,
but no system address. `markets.system_address` was `NOT NULL` behind a
foreign key onto `systems`, so a market for a system we have not
ingested had nowhere to go. That is #32.

Hold the name until the address is known. Both foreign keys on `markets`
are MATCH SIMPLE, so a null address satisfies them and the row waits.
Writing a system then links whatever named it, creating the station on
the way, so there is no sweep to run.

Markets move, too. Fleet carriers jump, and 82 in my database are
already recorded in more than one system, so where a market is now
replaces where it was rather than filling a blank.

Market data says only that a station is there, so an existing station is
left untouched. Its `updated_at` and `updated_by` keep pointing at
whoever last described the place.

Also drops `galos_db/.sqlx`. There were two query caches and sqlx
prefers the crate-local one, which had not been regenerated since 2024.
That is why the tests failed on CI and not locally.

Adds migration `20260802080000_accept_market_data`. Nothing
runs migrations on startup.

Fixes #32

- [x] Create stations on market data events
- [x] Don't update station when it already exists
- [x] Record markets whose system does not exist yet

nixpulvis committed:
Make station's ty field nullable

nixpulvis committed:
Create stations with market EDDN messages

fixes #32

nixpulvis committed:
Create stations with market EDDN messages

fixes #32

nixpulvis committed:
Show error when market event system doesn't exist

nixpulvis committed:
Show error when market event system doesn't exist

nixpulvis commented:
@mattyoung101 in case this means anything to you at this point. I think I've fixed this issue. I'm not loving the code at this point and think it goes to be time to refactor the DB code so all the creating dependent data is consistent, but it works for now.

I'll have to give some real thought to how to make this more ergonomic and assist with create/updating multiple structures at once. At one point I was thinking about rewriting everything to use [SeaORM](https://docs.rs/sea-orm/latest/sea_orm), but I'm pretty sure I've decided against that.

Anyway, just throwing a note here for you in case it helps.

nixpulvis commented on src/bin/galos-sync/eddn.rs:
```diff
@@ -219,9 +219,27 @@ fn process_message(db: &Database, message: Message, user: String) {
-                Err(err) => eprintln!("[EDDN] <MKT:mkt> {}", err),
+            ) => match System::fetch_by_name(db, &m.system_name).await {
+                Ok(system) => {
+                    if let Ok(_) = Station::create(
```
This line contains the fix.

nixpulvis commented:
I think I'd like to change this so that the station is **not** updated when it's market data is updated. At the moment we simple call `galos_db::Station::create` which also updates.

nixpulvis committed:
Update SQLX prepared data for testing

nixpulvis committed:
Update SQLX prepared data for testing

nixpulvis committed:
Update listings SQL example to return useful data

nixpulvis committed:
Update listings SQL example to return useful data

mlyoung101 commented:
@nixpulvis Hey thank you for the ping! This is great to see, and definitely helpful, I need to improve some of the querying stuff in my repo when I get some time.

I hear mixed things about ORMs; I am working on a uni assignment and using [Diesel](https://diesel.rs/) which has been quite nice, but is not async. I have read a lot that people recommend just using sqlx, but it sounds like you're running into a lot of complexity maintaining the queries manually?

Slightly unrelated to this specific issue, but although I haven't had much time to maintain EDTear due to thesis, I have been running it more or less continuously since February and have received about ~85 million trades. Given that the schema I use is basically your schema, I was wondering if you'd be interested in me writing an exporter of some sort for Galos? I would like to make some sort of automated service that dumps my Postgres database monthly, and then it might be nice to have a way to import that into Galos?

nixpulvis commented:
Given how quickly market data is outdated, I'm not sure how valuable syncing old data is tbh. It is an interesting question of how to handle outdated data and systems which are rarely visited. Or markets which were visited but the player isn't running an EDDN publisher.

For now I don't think I'm going to worry about it too much. If I ever really care, first I'd probably put the EDDN sync on a server rather than my laptop so I don't miss events.

nixpulvis commented:
I have no idea why the tests are failing on github actions and not locally. I'll have to dig into this more.

nixpulvis committed:
WIP

nixpulvis committed:
Drop the second, stale sqlx cache

CI type checks queries against the checked-in cache, and a DATABASE_URL
in the environment means local builds never read it, so the two could
disagree in silence.

There were two caches. sqlx prefers a crate-local .sqlx over the one at
the workspace root, and prepare has been writing to the root, leaving
galos_db/.sqlx untouched since 2024, still describing stations.ty as NOT
NULL from before it was made nullable here.

That is why the failure looked so odd: the station create query is new
and absent from the stale cache, so it fell through to the root one and
saw a nullable column, while from_journal was in the stale cache and saw
a non-null one. Two queries in one file answering to different schemas.

Keep the root cache, which already covers every query.

nixpulvis committed:
Record markets whose system we have never seen

The EDDN commodity schema gives systemName, stationName and marketId,
and no system address. So a market can arrive before anything that would
create the system it sits in, and issue #32 is that foreign key giving
way underneath it.

Hold the name until the address is known. Both foreign keys on markets
are MATCH SIMPLE, so a null address satisfies them and the row waits
without either constraint being weakened or a placeholder address being
invented.

The wait ends when the system turns up, rather than on a sweep: writing
a system asks whether any market named it, makes the station the market
needs, and links it. A partial index over just the waiting rows keeps
that question cheap enough to ask on every system write.

Fixes #32

nixpulvis committed:
Leave a station alone when it is already on record

Market data says only that the station is there. Restamping updated_at
and updated_by credited the freshness to whoever sold something rather
than to whoever last described the place.

DO NOTHING returns no row, so an existing station is read back instead.

nixpulvis committed:
Loosen the schema for market data

A market message knows less about where it is than a journal entry does.
It gives systemName, stationName and marketId, and no system address and
no station type, and the schema asked for both.

Station type becomes optional, and a market may name its system without
placing it, holding the name until the address is known.
